src/RealFourierArray.jl

```diff
@@ -10,6 +10,7 @@ import Base.eltype, Base.size, Base.length
 ## TODO this always allocates both real and fourier parts, which might be a bit wasteful
 ## TODO implement broadcasting. This seems to be non-trivial, so I just implement a few simple methods
 ## TODO implement compressed representation in Fourier space of real arrays
+## TODO generalize to AbstractArray
```

Comment:
Where do you need that to be an Array? Can you add a comment why it cannot be generalised yet?

Reply:
It definitely can, I've just been too lazy to figure out the correct dispatch here. In any case we'll have other troubles when we do get to non-Array types, so we'll see this when we need to

Reply:
Fixed it, needed it for passing tests anyway

Reply:
Sure I agree we are not ready for non-array types, I just wondered why you decided to make it less abstract.

Reply:
For performance. It killed the for loops operating on RFAs